Source/WebKit/GPUProcess/media/RemoteMediaPlayerProxy.cpp

```diff
@@ -697,13 +694,11 @@ void RemoteMediaPlayerProxy::mediaPlayerDidAddAudioTrack(WebCore::AudioTrackPriv
 
 void RemoteMediaPlayerProxy::mediaPlayerDidRemoveAudioTrack(WebCore::AudioTrackPrivate& track)
 {
-    ASSERT(m_audioTracks.contains(&track));
-    if (!m_audioTracks.contains(&track))
-        return;
-
-    auto audioTrack = m_audioTracks.get(&track);
-    m_webProcessConnection->send(Messages::MediaPlayerPrivateRemote::RemoveRemoteAudioTrack(audioTrack->identifier()), m_id);
-    m_audioTracks.remove(&track);
+    m_webProcessConnection->send(Messages::MediaPlayerPrivateRemote::RemoveRemoteAudioTrack(track.id()), m_id);
+    for (size_t index = 0; index < m_audioTracks.size(); index++) {
+        if (m_audioTracks[index]->id() == track.id())
+            m_audioTracks.remove(index);
+    }
```

Comment:
Vector already has a `removeFirstMatching(...)` function that makes this for loop unnecessary.